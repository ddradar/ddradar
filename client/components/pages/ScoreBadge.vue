<template>
  <b-tooltip :label="clearStatus">
    <span
      class="tag is-light"
      :class="{
        'is-white': lamp === 7,
        'is-basic': lamp === 6,
        'is-expert': lamp === 5,
        'is-beginner': lamp === 4,
        'is-difficult': lamp === 3,
        'is-challenge': lamp === 1,
        'is-light': lamp === 0,
      }"
    >
      {{ score }}
    </span>
  </b-tooltip>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'nuxt-property-decorator'

import { ClearLamp, clearLampList } from '~/types/api/score'

@Component
export default class ScoreBadgeComponent extends Vue {
  @Prop({ required: true, type: Number })
  readonly lamp!: ClearLamp

  @Prop({ required: true, type: Number })
  readonly score!: number

  get clearStatus() {
    return clearLampList[this.lamp]
  }
}
</script>
