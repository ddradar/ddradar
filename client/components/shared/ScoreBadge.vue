<template>
  <b-tooltip :label="clearStatus">
    <span class="tag" :class="tagClass">{{ score }}</span>
  </b-tooltip>
</template>

<script lang="ts">
import { Score } from '@ddradar/core'
import type { PropType } from '@nuxtjs/composition-api'
import { computed, defineComponent } from '@nuxtjs/composition-api'

const lampClassMap = new Map([
  [0, 'is-dark'],
  [1, 'is-challenge'],
  [2, 'is-primary'],
  [3, 'is-difficult'],
  [4, 'is-beginner'],
  [5, 'is-expert'],
  [6, 'is-basic'],
  [7, 'is-white'],
] as const)

export default defineComponent({
  props: {
    lamp: { type: Number as PropType<Score.ClearLamp>, required: true },
    score: { type: Number, required: true },
  },
  setup(props) {
    // Computed
    const clearStatus = computed(() => Score.clearLampMap.get(props.lamp))
    const tagClass = computed(() => lampClassMap.get(props.lamp))

    return { clearStatus, tagClass }
  },
})
</script>
