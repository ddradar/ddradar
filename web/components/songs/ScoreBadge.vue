<template>
  <OTooltip :label="clearLampMap.get(lamp)">
    <span class="tag" :class="classMap.get(lamp)">{{ score }}</span>
  </OTooltip>
</template>

<script lang="ts" setup>
import type { Score } from '@ddradar/core'
import { clearLampMap } from '@ddradar/core'

interface ScorebadgeProps {
  lamp: Score['clearLamp']
  score: Score['score']
}
const classMap = new Map([
  [0, 'is-dark'],
  [1, 'is-challenge'],
  [2, 'is-primary'],
  [3, 'is-difficult'],
  [4, 'is-beginner'],
  [5, 'is-expert'],
  [6, 'is-basic'],
  [7, 'is-white'],
] as const)

defineProps<ScorebadgeProps>()
</script>
