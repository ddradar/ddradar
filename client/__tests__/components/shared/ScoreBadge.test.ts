import { createLocalVue, mount } from '@vue/test-utils'
import Buefy from 'buefy'

import ScoreBadge from '~/components/shared/ScoreBadge.vue'

const localVue = createLocalVue()
localVue.use(Buefy)

describe('/components/shared/ScoreBadge.vue', () => {
  test.each([
    [0, 10000],
    [1, 340000],
    [2, 800000],
    [3, 950000],
    [4, 940000],
    [5, 987650],
    [6, 999990],
    [7, 1000000],
  ])('{ lamp: %i, score: %i } renders correctly', (lamp, score) => {
    // Arrange - Act
    const propsData = { lamp, score }
    const wrapper = mount(ScoreBadge, { localVue, propsData })

    // Assert
    expect(wrapper).toMatchSnapshot()
  })
})
