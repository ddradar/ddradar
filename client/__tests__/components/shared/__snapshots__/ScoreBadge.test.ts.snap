// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/components/shared/ScoreBadge.vue { lamp: 0, score: 10000 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Failed</div></transition-stub><div class="tooltip-trigger"><span class="tag is-dark">10000</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 1, score: 340000 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Assisted Clear</div></transition-stub><div class="tooltip-trigger"><span class="tag is-challenge">340000</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 2, score: 800000 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Clear</div></transition-stub><div class="tooltip-trigger"><span class="tag is-primary">800000</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 3, score: 950000 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Life 4</div></transition-stub><div class="tooltip-trigger"><span class="tag is-difficult">950000</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 4, score: 940000 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Full Combo</div></transition-stub><div class="tooltip-trigger"><span class="tag is-beginner">940000</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 5, score: 987650 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Great Full Combo</div></transition-stub><div class="tooltip-trigger"><span class="tag is-expert">987650</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 6, score: 999990 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Perfect Full Combo</div></transition-stub><div class="tooltip-trigger"><span class="tag is-basic">999990</span></div></span>`;

exports[`/components/shared/ScoreBadge.vue { lamp: 7, score: 1000000 } renders correctly 1`] = `<span class="b-tooltip is-primary is-top is-medium"><transition-stub name="fade"><div class="tooltip-content" style="display: none;">Marvelous Full Combo</div></transition-stub><div class="tooltip-trigger"><span class="tag is-white">1000000</span></div></span>`;
